### DISCUSSION TITLE
Upgrade to Mobx 6 from version 5

### DISCUSSION THREAD

**Discussion by besLisbeth** [mobxjs/mobx#2538] [Q&A] [answered]
Hi there 👋,
I've got some problems with the upgrade to version 6. All my @action.bound shows that `this` is undefined. I use them in the classes in the next construction: 
```
export default class Store {
    @observable value1 = [];
    @observable value2 = [];

    @action.bound
    myFn() {
        this.anotherFnCall(); // <- this is undefined here
    }
}
```
Does anyone also have any issues with them?

**Comment by danielkcz** [comment, marked as answer]
Please follow the issue template which also mentions the migration guide :/

**Comment by besLisbeth** [comment]
@FredyC, I'm really sorry, but I didn't find anything which could cause such an issue in the new migration guide or CHANGELOG. Was that caused by strict mode?

**Comment by mweststrate** [comment]
I'm not sure how you can have read all those in < 7 minutes...

https://mobx.js.org/migrating-from-4-or-5.html#upgrading-classes-to-use-makeobservable

**Reply by besLisbeth** [reply]
Thank you a lot, I really blame myself for not reading it up attentively. 
But, maybe, it will be helpful to also add warning to the console, as the errors I got didn't look like the problem with decorators? 
Thanks, anyway.

**Comment by mweststrate** [comment]
we would if we could. But it is hard to present things not being present :)
